lib/integrations/llm_base_service.rb

```diff
@@ -82,64 +81,84 @@ def event_is_cacheable?
     self.class::CACHEABLE_EVENTS.include?(event_name)
   end
 
-  def api_url
+  def api_base
     endpoint = InstallationConfig.find_by(name: 'CAPTAIN_OPEN_AI_ENDPOINT')&.value.presence || 'https://api.openai.com/'
     endpoint = endpoint.chomp('/')
-    "#{endpoint}/v1/chat/completions"
+    "#{endpoint}/v1"
   end
 
   def make_api_call(body)
     parsed_body = JSON.parse(body)
     instrumentation_params = build_instrumentation_params(parsed_body)
 
     instrument_llm_call(instrumentation_params) do
-      execute_api_request(body, parsed_body['messages'])
+      execute_ruby_llm_request(parsed_body)
     end
   end
 
-  def build_instrumentation_params(parsed_body)
-    {
-      span_name: "llm.#{event_name}",
-      account_id: hook.account_id,
-      conversation_id: conversation&.display_id,
-      feature_name: event_name,
-      model: parsed_body['model'],
-      messages: parsed_body['messages'],
-      temperature: parsed_body['temperature']
-    }
-  end
+  def execute_ruby_llm_request(parsed_body)
+    messages = parsed_body['messages']
+    model = parsed_body['model']
 
-  def execute_api_request(body, messages)
-    Rails.logger.info("OpenAI API request: #{body}")
-    response = HTTParty.post(api_url, headers: api_headers, body: body)
-    Rails.logger.info("OpenAI API response: #{response.body}")
+    Llm::Config.with_api_key(hook.settings['api_key'], api_base: api_base) do |context|
+      chat = context.chat(model: model)
+      setup_chat_with_messages(chat, messages)
+    end
+  rescue RubyLLM::Error => e
+    build_error_response_from_exception(e, messages)
```

Comment:
It seems like this error is important information... Let's also capture this exception, use the `ChatwootExceptionTracker` we can also set an alert on this. 

Let me know if I am missing something